packages/cursorless-engine/src/nodeCommon/TalonSpokenFormsJsonReader.ts

```diff
@@ -0,0 +1,86 @@
+import { Disposer, FileSystem, Notifier } from "@cursorless/common";
+import { mkdir, readFile } from "fs/promises";
+
+import * as path from "path";
+import {
+  NeedsInitialTalonUpdateError,
+  SpokenFormEntry,
+  TalonSpokenForms,
+} from "../scopeProviders/TalonSpokenForms";
+
+interface TalonSpokenFormsPayload {
+  version: number;
+  spokenForms: SpokenFormEntry[];
+}
+
+const LATEST_SPOKEN_FORMS_JSON_VERSION = 0;
+
+export class TalonSpokenFormsJsonReader implements TalonSpokenForms {
+  private disposer = new Disposer();
+  private notifier = new Notifier();
+
+  constructor(private fileSystem: FileSystem) {
+    this.init();
+  }
+
+  private async init() {
+    const parentDir = path.dirname(
+      this.fileSystem.cursorlessTalonStateJsonPath,
+    );
+    await mkdir(parentDir, { recursive: true });
```

Comment:
do this in extension init. might be able to remove some of this async stuff then too. and then maybe disposer could turn into function?